### PR TITLE
[Optimization][datasource registration] datasource registration should can save config before validate

### PR DESCRIPTION
**Pull request by wmtbnbo**
…d can save config before validate

## Purpose of the pull request

datasource registration should can save config before validate
[Optimization][datasource registration] datasource registration should can save config before validate https://github.com/DataLinkDC/dinky/issues/1255

## Verify this pull request

This pull request is code cleanup without any test coverage.

*(or)*

This pull request is already covered by existing tests, such as *(please describe tests)*.

(or)

This change added tests and can be verified as follows:

### CODE REVIEW

**Comment by aiwenmo** [comment]
Why is it modified in this way?

**Comment by wmtbnbo** [comment]
> Why is it modified in this way?

In this way, the capture of exceptions only affects the saving of the configuration, does not affect other heartbeat detection judgments, and can also catch exceptions caused by loading the driver when saving

File: dlink-admin/src/main/java/com/dlink/service/impl/DataBaseServiceImpl.java [hunk 1]
```diff
@@ -75,7 +75,11 @@ public boolean saveOrUpdateDataBase(DataBase dataBase) {
             return false;
         }
         if (Asserts.isNull(dataBase.getId())) {
-            checkHeartBeat(dataBase);
+            try {
+                checkHeartBeat(dataBase);
+            } catch (Exception e) {
+                throw new RuntimeException(e);
```
Comment (leechor) [inline dlink-admin/src/main/java/com/dlink/service/impl/DataBaseServiceImpl.java:81]: Does throw exception still prevent code from saving function? I thought the PR want to keep the save function running all the time.

Comment (wmtbnbo) [reply dlink-admin/src/main/java/com/dlink/service/impl/DataBaseServiceImpl.java:81]: Yes, when your configuration information is abnormal, it should not affect the saving of the data source, but my pr has a problem. I need to modify it again

**Pull request closed without merging**